cli/azd/pkg/httputil/poll_record.go

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+//go:build record
+
+package httputil
+
+import (
+	"os"
+	"time"
+)
+
+// By default, PollDelay is a no-op function that returns the delay as-is.
+// This function is overridden in the record mode to return the delay as 0.
+func PollDelay(delay time.Duration) time.Duration {
+	d := os.Getenv("AZD_TEST_POLL_DELAY")
```

Comment:
This deviates from our existing handling of LROs in recording (see `poll_discard.go`). I think we're at a point where it's too expensive to implement response trimmers for these non-standard poll operations, where we have business logic looking for specific fields in the response body.

It does unfortunately mean larger recording files, and more noise within the recorded interactions, but maybe that's okay.

Also open to any ideas here...

Reply:
Seems like a good approach - I suspect that @hemarina might want to leverage this technique with her AppService changes which are also logically polling an endpoint to check status.

I guess long term we could consider something like the recorder keeping a map of mutators that get to run over the recording snapshot before we save it so they could clean up (or elide) intermediate recordings, but I think this is a fine solution for now.

Reply:
@hemarina's change for deployment status uses an actual LRO / Azure LRO, our existing implementation will automatically handle it which is nice.

The part that gets trickier is for these types of desired-state polling with no unique operation IDs.

> a map of mutators that get to run over the recording snapshot before we save it so they could clean up (or elide) intermediate recordings

This looks pretty close to the existing poll_discard implementation. The thing that was concerning is scaling it to all the different types of desired states. For example, what if two parts of the app was polling for different end states. Without a unique operation ID, it would be hard to track.

As I am writing this comment, I suppose if we wanted to really implement it, we could smuggle a operation ID in record mode and we can automatically trim all except the last response.. Not a bad idea really...

Reply:
With the latest change, we call `httputil.PollHeader` which returns a unique polling ID in record mode, and simply `nil` in non-record mode. The responses are then trimmed by the recorder. I think this works out pretty well.